### PR TITLE
2.1.4 dataproduct agnostic

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

**Comment by gdgate** [comment]
Merge Failed.

This change or one of its cross-repo dependencies was unable to be automatically merged with the current state of its repository. Please rebase the change and upload a new patchset.

File: lib/gooddata/lcm/actions/collect_data_product.rb [hunk 1]
```diff
@@ -31,13 +31,21 @@ def call(params)
           domain = client.domain(domain_name) || fail("Invalid domain name specified - #{domain_name}")
 
           if params.key?(:data_product)
-            data_product_id = params.data_product
+            data_product = domain.data_products(params.data_product)
           else
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/collect_data_product.rb]: Calls 'params.gdc_logger' 2 times (DuplicateMethodCall)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/677/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/478/ : FAILURE in 2m 02s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/786/ : SUCCESS in 1m 54s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/678/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/479/ : FAILURE in 3m 16s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/787/ : SUCCESS in 2m 19s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/679/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/480/ : FAILURE in 51s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/788/ : SUCCESS in 38s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/680/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/481/ : SUCCESS in 45s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/789/ : SUCCESS in 35s

**Comment by panjan** [comment]
ok to merge

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/261/ : SUCCESS in 15s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/482/ : SUCCESS in 49s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/790/ : SUCCESS in 1m 51s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1179/ : FAILURE in 5m 23s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/937/ : ABORTED in 9m 29s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/262/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/483/ : SUCCESS in 57s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/791/ : SUCCESS in 34s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1180/ : FAILURE in 3m 03s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/938/ : ABORTED in 5m 41s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/263/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/484/ : SUCCESS in 55s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/792/ : SUCCESS in 34s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1181/ : FAILURE in 1h 00m 43s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/939/ : UNSTABLE in 38m 33s

**Comment by kubamahnert** [comment]
the build should pass from the appstore side as it requires a different version of tests than the one present in lcm2 (default picked branch)

**Pull request merged**